x-pack/plugins/osquery/cypress/integration/trusted_apps_pre.spec.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  TRUSTED_APPLICATIONS_FLYOUT_NAME_FIELD,
+  TRUSTED_APPLICATIONS_FLYOUT_DESCRIPTION_FIELD,
+  TRUSTED_APPLICATIONS_FLYOUT_CONDITIONS_BUILDER_FIELD,
+  TRUSTED_APPLICATIONS_FLYOUT_CONDITIONS_BUILDER_FIELD_PATH_OPTION,
+  TRUSTED_APPLICATIONS_FLYOUT_CONDITIONS_BUILDER_VALUE,
+} from '../screens/trusted_apps';
+import { FLEET_AGENT_POLICIES, navigateTo } from '../tasks/navigation';
+import { goToTrustedApplicationsList } from '../tasks/trusted_apps';
+import { login } from '../tasks/login';
+import {
+  ADD_POLICY_BTN,
+  CREATE_PACKAGE_POLICY_SAVE_BTN,
+  DATA_COLLECTION_SETUP_STEP,
+  PACKAGE_POLICY_NAME_INPUT,
+  PACKAGE_POLICY_DESCRIPTION_INPUT,
+} from '../screens/integrations';
+import { closeModalIfVisible } from '../tasks/integrations';
+
+const INTEGRATION_PACKAGE_NAME = 'Endpoint Security';
+
+const AGENT_POLICY_NAME = 'Agent policy from automation';
+
+const INTEGRATION_PACKAGE_POLICY_NAME = 'Endpoint integration';
+
+describe('Configure trusted apps before upgrade', () => {
+  before(() => {
+    login();
+    addAndVerifyAgentPolicy();
+  });
+
+  it('Add legacy trusted app', () => {
+    addAndVerifyIntegration();
```

Comment:
Could this be moved to the `before` clause just right after the `addAndVerifyAgentPolicy` ?